react-components/hello-world.md

```diff
@@ -0,0 +1,386 @@
+$ cd hello-world
+```
+
+## Running A Server
+
+Let's look at what the `create-react-app` tool has given us!
+
+As part of using Create React App, we'll begin to use the `yarn` CLI tool. With `yarn` CLI, we'll be able to run scripts that are defined in `package.json`, a file also generated by Create React App. These scripts can run servers that load and reload our React app, which simplifies our development workflow.
+
+In our project root, we start a server with this command in the terminal:
+
+```bash
+$ yarn start
+```
+
+By default, our front-end is viewable at `localhost:3000`.
+
+Every time we make a change in a file and save it, our server will detect changes and then reload the app immediately.
+
+![](../assets/react-components_hello-world-landing.png)
+
+<!-- Question 1 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: short-answer
+* id: 0b2e499f
+* title: Hello, World!
+##### !question
+
+In one sentence, describe the _UI elements_ you see on `localhost:3000`.
+
+##### !end-question
+##### !answer
+
+/.+/
+
+##### !end-answer
+##### !explanation
+
+An example list of visible UI elements include:
+
+- One spinning logo ⚛️ in the center
+- The text `Edit src/App.js and save to reload.`
+- A link labeled "Learn React" that directs us to the React website
+
+##### !end-explanation
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+## Inspecting the Code
+
+Our project should have the following structure:
+
+```
+├── public
+│   └── index.html
+├── src
+│   ├── App.css
+│   ├── App.js
+│   ├── App.test.js
+│   ├── index.css
+│   ├── index.js
+│   ├── logo.svg
+│   ├── reportWebVitals.js
+│   └── setupTests.js
+├── package.json
+├── README.md
+└── yarn.lock
+```
+
+In this curriculum, we will intentionally not cover the following files:
+
+- The contents of `public` besides `index.html`
+- `logo.svg`
+- `reportWebVitals.js`
+- `setupTests.js`
+
+There are two files that are the most important to understand when developing React for the first time:
+
+1. `src/index.js`
+1. `src/App.js`
+
+### Exploring `index.js`
+
+Let's go ahead and open this project up in our text editor to examine the files that have been created for us. We'll begin with `src/index.js` and practice our reading skills.
+
+```js
+import React from "react";
+import ReactDOM from "react-dom";
+import "./index.css";
+import App from "./App";
+import reportWebVitals from "./reportWebVitals";
+
+ReactDOM.render(
+  <React.StrictMode>
+    <App />
+  </React.StrictMode>,
+  document.getElementById("root")
+);
+
+// If you want to start measuring performance in your app, pass a function
+// to log results (for example: reportWebVitals(console.log))
+// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
+reportWebVitals();
+```
+
+<!-- Question 2 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: checkbox
+* id: e447ec71
+* title: Hello, World!
+##### !question
+
+`src/index.js` imports several files. Select the files that are imported.
```

Comment:
some of the imports are only getting certain types out of the files, so go with a more inclusive term
```suggestion
`src/index.js` imports several resources. Select the resources that are imported.
```